src/main/java/com/tradeshift/service/DanishHelloService.java

```diff
@@ -0,0 +1,12 @@
+package com.tradeshift.service;
+
+public class DanishHelloService implements HelloService {
+
+    @Override
+    public String formatName(String name) throws NullPointerException {
+        if(name == null || name.trim().length() == 0) {
+            throw new NullPointerException("Name can not be null or empty");
```

Comment:
This isn't a `NullPointerException`, if `name` is only whitespace, there are no nulls involved at all, but the exception will still be thrown. If you want to do parameter checking like this, you should be throwing another exception, like for example `IllegalArgumentException` that tells the caller what the problem actually is.